10-dns-bootstrap.md

```diff
@@ -1,61 +1,62 @@
+This specification describes a node discovery mechanism based on the Domain Name System (DNS).
 Its purpose is twofold:
 
- - Bootstrap: the initial node discovery for nodes that have no known contacts in the network
- - Assisted Node Location: supporting nodes to discover the current network address of previously known peers
+ - Bootstrap: providing the initial node discovery for nodes that have no known contacts in the network
+ - Assisted Node Location: supporting nodes in  discovery of the current network address of previously known peers
 
-A domain name server implementing this specification is called a _DNS Seed_, and answers incoming DNS queries of type `A`, `AAAA` or `SRV` as specified in RFCs 1035<sup>[1](#ref-1)</sup>, 3596<sup>[2](#ref-2)</sup> and 2782<sup>[3](#ref-3)</sup> respectively.
+A domain name server implementing this specification is called a _DNS Seed_, and answers incoming DNS queries of type `A`, `AAAA`, or `SRV` as specified in RFCs 1035<sup>[1](#ref-1)</sup>, 3596<sup>[2](#ref-2)</sup> and 2782<sup>[3](#ref-3)</sup> respectively.
 The DNS server is authoritative for a subdomain, called a _seed root domain_, and clients may query it for subdomains.
 
 The subdomains consist of a number of dot-separated _conditions_ that further narrow the desired results.
 
 ## DNS Seed Queries
 
-A client MAY issue queries using either the `A`, `AAAA` or `SRV` query types, specifying conditions for the desired results the seed should return.
+A client MAY issue queries using the `A`, `AAAA`, or `SRV` query types, specifying conditions for the desired results that the seed should return.
 
 ### Query Semantics
 
-The conditions are key-value pairs with a single letter key and the remainder as the value.
+The conditions are key-value pairs with a single-letter key; the remainder of the key-value pair is the value.
 The following key-value pairs MUST be supported by a DNS seed:
 
  - `r`: realm byte, used to specify what realm the returned nodes must support (default value: 0, Bitcoin)
- - `a`: address types, used to specify what address types should be returned for `SRV` queries. This is a bitfield using the types from [BOLT 7](07-routing-gossip.md) as bit index. This condition MAY only be used for `SRV` queries. (default value: 6, i.e., `2 || 4`, IPv4 and IPv6)
- - `l`: `node_id`, the bech32 encoded `node_id` of a specific node, used to ask for a single node instead of a random selection. (default: null)
- - `n`: textual representation of the number of desired reply records (default: 25)
+ - `a`: address types, used to specify what address types should be returned for `SRV` queries. This is a bitfield that uses the types from [BOLT 7](07-routing-gossip.md) as bit index. This condition MAY only be used for `SRV` queries. (default value: 6, i.e., `2 || 4`, IPv4 and IPv6) [ FIXME: BOLT 7 says that IPv4 and IPv6 are address descriptor types 1 & 2. Are we missing something in this description? ]
```

Comment:
Ah, this is subtle.  The 2 == "bit 1" and 4 == "bit 2" (counting bits from 0).   Perhaps expand:

(default value: 6, ie. bit 1 and bit 2 set for IPv4 and IPv4 respectively)

?

Reply:
Yes, that was the intention, sorry for burrying the lead :-)